### PR TITLE
Slack app settings

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/19435

Adds UI to configure slack app.

Screenshot 2022-01-10 at 14 37 52

How to test:
- Merge https://github.com/metabase/metabase/pull/19592 locally into this branch
- Make sure it is possible to create a slack app following the instructions in Admin > Slack

### CODE REVIEW

File: frontend/src/metabase/admin/app/components/AdminApp/AdminApp.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from "react";
+import DeprecationBanner from "../../containers/DeprecationNotice";
```
Comment (alxnddr) [inline frontend/src/metabase/admin/app/components/AdminApp/AdminApp.tsx:2]: Probably, you missed this import when renamed the component `DeprecationBanner ` -> `DeprecationNotice`

File: frontend/src/metabase-types/api/settings.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export interface Version {
+  tag: string;
+}
+
+export interface Settings {
```
Comment (alxnddr) [inline frontend/src/metabase-types/api/settings.ts:5]: It would be nice to use these types in `frontend/src/metabase/lib/settings.ts`

**Review by gusaiani — APPROVED** [review]
Pretty good stuff.

I saw the happy path work, plus adding wrong oAuth.

Some bugbashing may catch things we missed, but looks good for a merge.

Beautiful work, Aleksandr.

File: frontend/src/metabase/admin/app/components/AdminApp/AdminApp.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/admin/app/components/AdminApp/AdminApp.tsx]: Yep, fixed

**Pull request merged**